packages/dapp-toolkit/src/modules/wallet-request/request-items/request-item.module.ts

```diff
@@ -38,21 +47,36 @@ export const RequestItemModule = (input: RequestItemModuleInput) => {
     isOneTimeRequest,
   })
 
-  const add = (value: {
-    type: RequestItem['type']
-    walletInteraction: WalletInteraction
-    isOneTimeRequest: boolean
-  }) => {
+  const add = (
+    value: {
+      type: RequestItem['type']
+      walletInteraction: WalletInteraction
+      isOneTimeRequest: boolean
+    },
+    onSignal?: (signalValue: string) => void,
+  ) => {
     const item = createItem(value)
     logger?.debug({
       method: 'addRequestItem',
       item,
     })
+    if (onSignal) {
+      signals.set(item.interactionId, onSignal)
+    }
+
     return storageModule
       .setItems({ [item.interactionId]: item })
       .map(() => item)
   }
 
+  const maybeGetSignal = (interactionId: string) => {
```

Comment:
Make this fn name more descriptive  

`type Signal = (txId: string) => void`

`getAndRemoveSignal: (interactionId: string) => Signal | undefined`